test/e2e/push_test.go

```diff
@@ -141,6 +141,44 @@ var _ = Describe("Podman push", func() {
 		Expect(output).To(ContainSubstring("zstd"))
 	})
 
+	It("push test --force-compression --format=v2s2", func() {
+		if podmanTest.Host.Arch == "ppc64le" {
+			Skip("No registry image for ppc64le")
+		}
+		if isRootless() {
+			err := podmanTest.RestoreArtifact(REGISTRY_IMAGE)
+			Expect(err).ToNot(HaveOccurred())
+		}
+		lock := GetPortLock("5005")
+		defer lock.Unlock()
+		session := podmanTest.Podman([]string{"run", "-d", "--name", "registry", "-p", "5005:5000", REGISTRY_IMAGE, "/entrypoint.sh", "/etc/docker/registry/config.yml"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		if !WaitContainerReady(podmanTest, "registry", "listening on", 20, 1) {
+			Skip("Cannot start docker registry.")
+		}
+
+		session = podmanTest.Podman([]string{"build", "-t", "imageone", "build/basicalpine"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		push := podmanTest.Podman([]string{"push", "-q", "--tls-verify=false", "--force-compression=true", "--compression-format", "gzip", "--format", "v2s2", "--remove-signatures", "imageone", "localhost:5005/image"})
+		push.WaitWithDefaultTimeout()
+		Expect(push).Should(ExitCleanly())
+
+		skopeoInspect := []string{"inspect", "--tls-verify=false", "--raw", "docker://localhost:5005/image:latest"}
+		skopeo := SystemExec("skopeo", skopeoInspect)
+		skopeo.WaitWithDefaultTimeout()
+		Expect(skopeo).Should(ExitCleanly())
+		output := skopeo.OutputToString()
+		Expect(output).ToNot(ContainSubstring("zstd"))
+
+		push = podmanTest.Podman([]string{"push", "-q", "--tls-verify=false", "--force-compression=true", "--compression-format", "zstd", "--format", "v2s2", "--remove-signatures", "imageone", "localhost:5005/image"})
+		push.WaitWithDefaultTimeout()
+		Expect(push).Should(ExitWithError(125, "cannot use ForceCompressionFormat with undefined default compression format"))
```

Comment:
This error seems extremely unhelpful. Why `undefined default compression format`? The command clearly specified zstd as format so this makes just no sense.

Reply:
Yes, that's just testing that the change to common doesn't mean that other invalid cases will still be attempted.

The fix to common is only to allow it to work in the gzip case, not to fix the misleading error completely.

Reply:
Yes, this is all a fallout of the c/common code being a short-term hack (where we didn’t do the long-term correct thing), see https://github.com/containers/common/pull/1869 .

What we _want_ is to differentiate between compression formats coming from `containers.conf` (which should be silently ignored for destinations which can’t support them, silently falling back to `gzip`) and compression formats coming from the command line (which should cause an error when the user asks for an impossible operation).

But the podman CLI (here https://github.com/containers/podman/blob/ceee7cb0a6b92412f6be5d6206cd4ac764cd99e6/cmd/podman/images/push.go#L140 ) is in the habit of using the `containers.conf` values as CLI option defaults, making it, from the very start, hard to tell the difference (although, IIRC, possible). And by the time the data gets to c/common, there is no longer any way to tell, in the current c/common API.

---

I do think that a comment here, in the test case, to the effect of “the command is asking for an impossible thing; per https://github.com/containers/common/pull/1869, we should detect that and fail with a precise error, but that does not exist, so we end up reporting this misleading text” would be useful.